src/all/mangafire/src/eu/kanade/tachiyomi/extension/all/mangafire/MangaFire.kt

```diff
@@ -147,137 +150,115 @@ class MangaFire(
     }
 
     private fun mangaDetailsParse(document: Document) = SManga.create().apply {
-        val root = document.selectFirst(".info")!!
-        val mangaTitle = root.child(1).ownText()
-        title = mangaTitle
-        description = document.run {
-            val description = selectFirst(Evaluator.Class("description"))!!.ownText()
-            when (val altTitle = root.child(2).ownText()) {
-                "", mangaTitle -> description
-                else -> "$description\n\nAlternative Title: $altTitle"
+        with(document.selectFirst(".main-inner:not(.manga-bottom)")!!) {
+            title = selectFirst("h1")!!.text()
+            thumbnail_url = selectFirst(".poster img")!!.attr("src")
+            status = selectFirst(".info > p").parseStatus()
+            description = buildString {
+                document.selectFirst("#synopsis .modal-content")?.textNodes()?.let {
+                    append(it.joinToString("\n\n"))
+                }
+
+                selectFirst("h6")?.let {
+                    append("\n\nAlternative title: ${it.text()}")
+                }
+            }.trim()
+
+            with(selectFirst(".meta")!!) {
+                author = selectFirst("span:contains(Author:) + span")?.text()
+                val type = selectFirst("span:contains(Type:) + span")?.text()
+                val genres = selectFirst("span:contains(Genres:) + span")?.text()
+                genre = listOfNotNull(type, genres).joinToString()
             }
```

Comment:
Since this is only used for non-critical fields it should not stop processing if not found.